src/input/stream/Streamer.cpp

```diff
@@ -105,21 +105,38 @@ bool Streamer::isDataAvailable() {
 }
 
 bool Streamer::readFullTSPacket(mpegts::PacketBuffer &buffer) {
-	if (_udpMultiListen.getFD() != -1) {
+	if (_udpMultiListen.getFD() == -1) {
+		return false;
+	}
+	do {
```

Comment:
Is this do-while loop constuction needed?

I still fail to understand the `buffer.markToFlush()`, I know you said it is more efficient.

I think this could be written as (you agree?):
```
  if (readSize > 0) {
    buffer.addAmountOfBytesWritten(readSize);
    buffer.trySyncing();
    // Add data to Filter
    _deviceData.getFilterData().addData(_feID, buffer, _deviceData.isInternalPidFilteringEnabled());
    return buffer.full();
  }
  return false
```
This is more straight forward and the buffer will be filled on next run as it always was.

Reply:
Hi @Barracuda09 ,

Yes, I feel the problem is that you don't get the reason for the function `buffer.markToFlush()`.
So please check this code in the current master:
https://github.com/Barracuda09/SATPI/blob/master/src/output/StreamThreadBase.cpp#L226-L234
This is executed in `StreamThreadBase::readDataFromInputDevice()` after calling to `inputDevice->readFullTSPacket(_tsBuffer[_writeIndex])`. And if the `_tsBuffer` not pass the check of `isReadyToSend()` then it never will be sended but discarded. Or I'm missing something?

Different from that, my block of `do { ... } while (1)` is becuase I want to fully read the Buffer (7 TS packets). If not then, I can re-read or flush it. The current implementation is based on the same code as for Child PIPE. However, I'll change it in the future when I'll add a re-read if more data is waiting in the incoming socket. I hope you agree with this.

But the main question is what happens with a non-full Buffer? If the buffer will wait for future reading, then I can reduce the complexity. But the check of `isReadyToSend()` will remove the data (empty the buffer) before returning to calling to `StreamThreadBase::readDataFromInputDevice()`, what I don't want. Any idea to fix or improve this?

Reply:
> https://github.com/Barracuda09/SATPI/blob/master/src/output/StreamThreadBase.cpp#L226-L234

This is only to send a empty (null) packet if the buffer is not full in 100 microseconds. So this should be no problem.